verifier/verifier.go

```diff
@@ -0,0 +1,494 @@
+			if err != nil {
+				return err
+			}
+			return processPluginResponse(capabilitiesToVerify, response, outcome)
+		}
+	}
+
+	return nil
+}
+
+func processPluginResponse(capabilitiesToVerify []proto.Capability, response *proto.VerifySignatureResponse, outcome *notation.VerificationOutcome) error {
+	verificationPluginName, err := getVerificationPlugin(&outcome.EnvelopeContent.SignerInfo)
+	if err != nil {
+		return err
+	}
+
+	// verify all extended critical attributes are processed by the plugin
+	for _, attr := range getNonPluginExtendedCriticalAttributes(&outcome.EnvelopeContent.SignerInfo) {
+		if !slice.ContainsAny(response.ProcessedAttributes, attr.Key) {
+			return fmt.Errorf("extended critical attribute %q was not processed by the verification plugin %q (all extended critical attributes must be processed by the verification plugin)", attr.Key, verificationPluginName)
+		}
+	}
+
+	for _, capability := range capabilitiesToVerify {
+		pluginResult := response.VerificationResults[capability]
+		if pluginResult == nil {
+			// verification result is empty for this capability
+			return notation.ErrorVerificationInconclusive{Msg: fmt.Sprintf("verification plugin %q failed to verify %q", verificationPluginName, capability)}
+		}
+		switch capability {
+		case proto.CapabilityTrustedIdentityVerifier:
+			if !pluginResult.Success {
+				// find the Authenticity VerificationResult that we already created during x509 trust store verification
+				var authenticityResult *notation.ValidationResult
+				for _, r := range outcome.VerificationResults {
+					if r.Type == trustpolicy.TypeAuthenticity {
+						authenticityResult = r
+						break
+					}
+				}
+
+				authenticityResult.Error = fmt.Errorf("trusted identify verification by plugin %q failed with reason %q", verificationPluginName, pluginResult.Reason)
+
+				if isCriticalFailure(authenticityResult) {
+					return authenticityResult.Error
+				}
+			}
+		case proto.CapabilityRevocationCheckVerifier:
+			var revocationResult *notation.ValidationResult
+			if !pluginResult.Success {
+				revocationResult = &notation.ValidationResult{
+					Error:  fmt.Errorf("revocation check by verification plugin %q failed with reason %q", verificationPluginName, pluginResult.Reason),
+					Type:   trustpolicy.TypeRevocation,
+					Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+				}
+			} else {
+				revocationResult = &notation.ValidationResult{
+					Type:   trustpolicy.TypeRevocation,
+					Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+				}
+			}
+			outcome.VerificationResults = append(outcome.VerificationResults, revocationResult)
+			if isCriticalFailure(revocationResult) {
+				return revocationResult.Error
+			}
+		}
+	}
+
+	return nil
+}
+
+func metadata(ctx context.Context, installedPlugin plugin.Plugin, pluginConfig map[string]string) (*proto.GetMetadataResponse, error) {
```

Comment:
Suggest renaming function name to `getPluginMetadata` and move this function to `helpers.go` since it is not directly related to verify.